cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/lineage/LineageOperationsProcessor.java

```diff
@@ -66,8 +67,8 @@ public class LineageOperationsProcessor {
   private final Dag stageDag;
   // Map of stage to list of operations recorded by that stage
   private final Map<String, List<FieldOperation>> stageOperations;
-  // Map of stage name to another map which contains the output field names to the corresponding origin
-  private final Map<String, Map<String, String>> stageOutputsWithOrigins;
+  // Map of stage name to another map which contains the output field names to their possible origins
+  private final Map<String, Map<String, List<Origin>>> stageOutputsWithOrigins;
```

Comment:
Should modify javadoc mentioning the tag is only useful for a read->write operations

Reply:
Added a note in `Origin` class